### ISSUE TITLE
pex.resolvable.InvalidRequirement: Unknown requirement type: .

### ISSUE THREAD

**Issue by wickman** [pantsbuild/pex#360]
It used to be possible to clone a python repository and do 'pex .' (in other words, check if path is an exploded source package) and build it.  Not sure when this behavior changed -- breaks some packaging scripts I've been using for periscope.

**Comment by wickman** [comment]
Worth noting: `FOO=$(basename $PWD); cd .. && pex $FOO` still works fine.

**Comment by jsirois** [comment]
Things work under 2.6, but fail as described under 2.7 and 3.6 at least:

+ 2.6
```
jsirois@gill ~ $ rm -rf /tmp/flask && git clone -q https://github.com/pallets/flask /tmp/flask && (cd /tmp/flask && PYTHONPATH=~/dev/pantsbuild/pex python2.6 -mpex.bin.pex .)
Python 2.6.9 (unknown, Mar 31 2016, 12:12:48) 
[GCC 5.3.0] on linux4
Type "help", "copyright", "credits" or "license" for more information.
(InteractiveConsole)
>>> import flask
>>> 
```

+ 2.7
```
jsirois@gill ~ $ rm -rf /tmp/flask && git clone -q https://github.com/pallets/flask /tmp/flask && (cd /tmp/flask && PYTHONPATH=~/dev/pantsbuild/pex python2.7 -mpex.bin.pex .)
Traceback (most recent call last):
  File "/usr/lib/python2.7/runpy.py", line 174, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "/usr/lib/python2.7/runpy.py", line 72, in _run_code
    exec code in run_globals
  File "/home/jsirois/dev/pantsbuild/pex/pex/bin/pex.py", line 599, in <module>
    main()
  File "/home/jsirois/dev/pantsbuild/pex/pex/bin/pex.py", line 578, in main
    pex_builder = build_pex(reqs, options, resolver_options_builder)
  File "/home/jsirois/dev/pantsbuild/pex/pex/bin/pex.py", line 504, in build_pex
    resolvables = [Resolvable.get(arg, resolver_option_builder) for arg in args]
  File "/home/jsirois/dev/pantsbuild/pex/pex/resolvable.py", line 71, in get
    raise cls.InvalidRequirement('Unknown requirement type: %s' % resolvable_string)
pex.resolvable.InvalidRequirement: Unknown requirement type: .
```

+ 3.6
```
jsirois@gill ~ $ rm -rf /tmp/flask && git clone -q https://github.com/pallets/flask /tmp/flask && (cd /tmp/flask && PYTHONPATH=~/dev/pantsbuild/pex python3.6 -mpex.bin.pex .)
Traceback (most recent call last):
  File "/usr/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/home/jsirois/dev/pantsbuild/pex/pex/bin/pex.py", line 599, in <module>
    main()
  File "/home/jsirois/dev/pantsbuild/pex/pex/bin/pex.py", line 578, in main
    pex_builder = build_pex(reqs, options, resolver_options_builder)
  File "/home/jsirois/dev/pantsbuild/pex/pex/bin/pex.py", line 504, in build_pex
    resolvables = [Resolvable.get(arg, resolver_option_builder) for arg in args]
  File "/home/jsirois/dev/pantsbuild/pex/pex/bin/pex.py", line 504, in <listcomp>
    resolvables = [Resolvable.get(arg, resolver_option_builder) for arg in args]
  File "/home/jsirois/dev/pantsbuild/pex/pex/resolvable.py", line 71, in get
    raise cls.InvalidRequirement('Unknown requirement type: %s' % resolvable_string)
pex.resolvable.InvalidRequirement: Unknown requirement type: .
```

**Comment by kwlzn** [comment]
different results on my machine for 2.7:

2.7:
{code}
[illuminati pex (master)]$ rm -rf /tmp/flask && git clone -q https://github.com/pallets/flask /tmp/flask && (cd /tmp/flask && PYTHONPATH=~/dev/pex python2.7 -mpex.bin.pex .)
Python 2.7.11 (default, Jan  5 2016, 18:06:15) 
[GCC 4.2.1 Compatible Apple LLVM 7.0.2 (clang-700.1.81)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
(InteractiveConsole)
>>> import flask; flask.__file__
'/private/var/folders/3t/xkwqrkld4xxgklk2s4n41jb80000gn/T/tmp5UMKoF/.deps/Flask-0.13.dev0-py2.py3-none-any.whl/flask/__init__.py'
>>>
{code}

3.6
{code}
[illuminati pex (master)]$ rm -rf /tmp/flask && git clone -q https://github.com/pallets/flask /tmp/flask && (cd /tmp/flask && PYTHONPATH=~/dev/pex python3.6 -mpex.bin.pex .)
Traceback (most recent call last):
  File "/Users/kwilson/Python/CPython-3.6.0/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/Users/kwilson/Python/CPython-3.6.0/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/Users/kwilson/dev/pex/pex/bin/pex.py", line 599, in <module>
    main()
  File "/Users/kwilson/dev/pex/pex/bin/pex.py", line 578, in main
    pex_builder = build_pex(reqs, options, resolver_options_builder)
  File "/Users/kwilson/dev/pex/pex/bin/pex.py", line 504, in build_pex
    resolvables = [Resolvable.get(arg, resolver_option_builder) for arg in args]
  File "/Users/kwilson/dev/pex/pex/bin/pex.py", line 504, in <listcomp>
    resolvables = [Resolvable.get(arg, resolver_option_builder) for arg in args]
  File "/Users/kwilson/dev/pex/pex/resolvable.py", line 71, in get
    raise cls.InvalidRequirement('Unknown requirement type: %s' % resolvable_string)
pex.resolvable.InvalidRequirement: Unknown requirement type: .
{code}

the failure appears related to the setuptools version:

{code}
[illuminati pex (master)]$ python2.7 -c 'import setuptools; print(setuptools.__version__)'
30.4.0
[illuminati pex (master)]$ python3.6 -c 'import setuptools; print(setuptools.__version__)'
34.2.0
[illuminati pex (master)]$ python3.6 -m pip install setuptools==30.4.0
Collecting setuptools==30.4.0
  Using cached setuptools-30.4.0-py2.py3-none-any.whl
Installing collected packages: setuptools
  Found existing installation: setuptools 34.2.0
    Uninstalling setuptools-34.2.0:
      Successfully uninstalled setuptools-34.2.0
Successfully installed setuptools-30.4.0
[illuminati pex (master)]$ rm -rf /tmp/flask && git clone -q https://github.com/pallets/flask /tmp/flask && (cd /tmp/flask && PYTHONPATH=~/dev/pex python3.6 -mpex.bin.pex .)
Python 3.6.0 (default, Feb 13 2017, 11:02:15) 
[GCC 4.2.1 Compatible Apple LLVM 8.0.0 (clang-800.0.42.1)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
(InteractiveConsole)
>>> import flask; flask.__file__
'/private/var/folders/3t/xkwqrkld4xxgklk2s4n41jb80000gn/T/tmp9n4cxpb3/.deps/Flask-0.13.dev0-py2.py3-none-any.whl/flask/__init__.py'
>>> 
{code}

**Comment by kwlzn** [comment]
seems to have started with 34.x.x

**Comment by habnabit** [comment]
Any suggestion on how to mitigate this? Just downgrade setuptools?

**Comment by kwlzn** [comment]
@habnabit yep, afaict a downgrade to `setuptools<34.0` should repair this.

**Comment by drujon** [comment]
Using the directory name worked for me without `cd ..`
in `/user/repo/foo`
`pex . -c foo.py -o foo.pex` fails
`pex foo -c foo.py -o foo.pex` works

**Comment by jsirois** [comment]
This has since been fixed with the vendoring of setuptools in #624 as tracked by #607.
